app/controllers/posts_controller.rb

```diff
@@ -345,7 +345,9 @@ def destroy_many
     posts.each { |p| guardian.ensure_can_delete!(p) }
 
     Post.transaction do
-      posts.each { |p| PostDestroyer.new(current_user, p, defer_flags: defer_flags).destroy }
+      posts.each_with_index do |p, i|
+        PostDestroyer.new(current_user, p, defer_flags: (agree_with_first_reply_flag.to_s == "true" && i == 0) ? false : true).destroy
```

Comment:
`agree_with_first_reply_flag.to_s == "true"` should be moved outside the loop so that this line becomes

``` ruby
PostDestroyer.new(current_user, p, defer_flags: !agree_with_first_reply_flag || i > 0).destroy
```